File: 2549. Count Distinct Numbers on Board/Solution.java
class Solution {
    public int distinctIntegers(int n) {
        Set<Integer> seen = new HashSet<>();
        seen.add(n);
        for (int i = n; i >= 1; i--) {
            for (int j = 1; j < n; j++) {
                if (i % j == 1) seen.add(j);
            }
        }
        return seen.size();
    }
}

File: 2549. Count Distinct Numbers on Board/Solution.py
class Solution:
    def distinctIntegers(self, n: int) -> int:
        seen = set()
        seen.add(n)
        for i in range(n, 0, -1):
            for j in range(1, n):
                if i % j == 1: seen.add(j)
        return len(seen)